staker/staker.go

```diff
@@ -280,8 +289,42 @@ func (s *Staker) Initialize(ctx context.Context) error {
 	return nil
 }
 
+func (s *Staker) latestNodeDetailsForUpdate(ctx context.Context, description string, node uint64) (arbutil.MessageIndex, *validator.GoGlobalState, error) {
+	stakedInfo, err := s.rollup.LookupNode(ctx, node)
```

Comment:
you could merge more code by using the fact that LatestConfirmed is identical to LatestStaked when getting zeros for address.
Could even test if walletOrZero is zeroes and in that case skip checking latestStaked (instead - use what you have for latestConfirmed)

Reply:
I've unified the functions. Let me know what you think. I spent a while thinking about how to do this and my current approach is the best I came up with, but there might be a slightly cleaner way.

Reply:
One thing I didn't want to do is have `LatestStakedNotifier` and `LatestConfirmedNotifier` be the same interface, because I could imagine some components wanting both, and I don't want them to be confusable (looking at a component it should be clear whether it gets the latest staked or confirmed info).